proxyserver/src/test/java/io/trino/gateway/proxyserver/TestProxyServer.java

```diff
@@ -79,13 +80,50 @@ public void testCustomHeader() throws Exception {
     }
   }
 
+  @Test
+  public void testLongHeader() throws Exception {
+    String mockResponseText = "CUSTOM LONG HEADER TEST";
+    String mockLongHeaderKey = "HEADER_LONG";
+    // Mockserver has max 8k for HTTP Header values so test with header value larger than default 4k
+    String mockLongHeaderValue = RandomStringUtils.random(5*1024, true, true);
+    int backendPort = 30000 + new Random().nextInt(1000);
+
+    MockWebServer backend = new MockWebServer();
+    backend.enqueue(new MockResponse().setBody(mockResponseText));
+    backend.play(backendPort);
+
+    int serverPort = backendPort + 1;
+    ProxyServerConfiguration config = buildConfig(backend.getUrl("/").toString(), serverPort);
+    ProxyServer proxyServer = new ProxyServer(config, new ProxyHandler());
```

Comment:
This code is used in several of the `TestProxyServer` tests - while you are in here could you extract it to a separate method?

Reply:
extracted to setProxyServer method